design-documents/graph-ql/coverage/gift-registry.graphqls

```diff
@@ -0,0 +1,316 @@
+}
+
+type GiftRegistrySelectAttributeOptionMetadata {
+    code: ID!
+    label: String!
+    is_default: Boolean
+}
+
+type GiftRegistry {
+    id: ID!
+    event_name: String!
+    type: GiftRegistryType
+    message: String!
+    created_on: String! @doc(description: "Creation date") @doc(description: "Accessible to the registry owner only")
+    privacy_settings: GiftRegistryPrivacySettings! @doc(description: "Accessible to the registry owner only")
+    status: GiftRegistryStatus! @doc(description: "Accessible to the registry owner only")
+    owner_name: String!
+    registrants: [GiftRegistryRegistrant]
+    shipping_address: CustomerAddress @doc(description: "Accessible to the registry owner only")
+    dynamic_attributes: [GiftRegistryDynamicAttribute]
+    items: [GiftRegistryItemInterface]
+}
+
+interface GiftRegistryItemInterface {
+    id: String!
+    quantity: Float!
+    quantity_fulfilled: Float!
+    note: String
+    added_on: String!
+    product: ProductInterface
+}
+
+type SimpleGiftRegistryItem implements GiftRegistryItemInterface {
+    customizable_options: [SelectedCustomizableOption]
+}
+
+type BundleGiftRegistryItem implements GiftRegistryItemInterface {
+    customizable_options: [SelectedCustomizableOption]
+    bundle_options: [SelectedBundleOption!]
+}
+
+type ConfigurableGiftRegistryItem implements GiftRegistryItemInterface {
+    customizable_options: [SelectedCustomizableOption]
+    configurable_options: [SelectedConfigurableOption!]
+}
+
+# Not currently supported by Magento core
+type DownloadableGiftRegistryItem implements GiftRegistryItemInterface {
+    customizable_options: [SelectedCustomizableOption]
+    links: [DownloadableProductLinks]
+    samples: [DownloadableProductSamples]
+}
+
+# Not currently supported by Magento core
+type VirtualGiftRegistryItem implements GiftRegistryItemInterface {
+    customizable_options: [SelectedCustomizableOption]
+}
+
+# Not currently supported by Magento core
+type GiftCardGiftRegistryItem implements GiftRegistryItemInterface {
+    sender_name: String!
+    recepient_name: String!
+    amount: Money!
+    message: String
+    customizable_options: [SelectedCustomizableOption]
+}
+
+interface GiftRegistryDynamicAttributeInterface  {
+    code: ID!
+    label: String!
+    value: String!
+}
+
+type GiftRegistryDynamicAttribute implements GiftRegistryDynamicAttributeInterface {
+    group: GiftRegistryDynamicAttributeGroup!
+}
+
+type GiftRegistryRegistrantDynamicAttribute implements GiftRegistryDynamicAttributeInterface {
+
+}
+
+enum GiftRegistryDynamicAttributeGroup {
+    GENERAL_INFORMATION
+    EVENT_INFORMATION
+    PRIVACY_SETTINGS
+    DETAILED_INFORMATION
+    SHIPPING_ADDRESS
+}
+
+type GiftRegistryRegistrant {
+    id: ID!
+    first_name: String!
+    last_name: String!
+    email: String! @doc(description: "Accessible to the registry owner only")
+    dynamic_attributes: [GiftRegistryRegistrantDynamicAttribute]
+}
+
+enum GiftRegistryStatus {
```

Comment:
I really like that we're using an enum for status instead of something like an `is_active` boolean 👍